frontend/libs/studio-components/src/components/StudioProfileMenu/StudioProfileMenu.tsx

```diff
@@ -0,0 +1,111 @@
+import React, { type ReactNode, type ReactElement, useState } from 'react';
+import classes from './StudioProfileMenu.module.css';
+import { Divider, DropdownMenu } from '@digdir/designsystemet-react';
+import { StudioPageHeaderButton } from '../StudioPageHeader';
+import { type StudioPageHeaderColor } from '../StudioPageHeader/types/StudioPageHeaderColor';
+import { type StudioPageHeaderVariant } from '../StudioPageHeader/types/StudioPageHeaderVariant';
+
+type StudioProfileMenuItemButton = {
+  type: 'button';
+  onClick: () => void;
+};
+
+type StudioProfileMenuItemLink = {
+  type: 'link';
+  href: string;
+  openInNewTab?: boolean;
+};
+
+export type StudioProfileMenuItem = {
+  action: StudioProfileMenuItemButton | StudioProfileMenuItemLink;
+  itemName: string;
+  isActive?: boolean;
+  hasDivider?: boolean;
+};
+
+export type StudioProfileMenuProps = {
+  triggerButtonText?: string;
+  profileImage: ReactNode;
+  profileMenuItems: StudioProfileMenuItem[];
+  color: StudioPageHeaderColor;
+  variant: StudioPageHeaderVariant;
+  ariaLabelTriggerButton: string;
+};
+
+export const StudioProfileMenu = ({
+  triggerButtonText,
+  profileImage,
+  profileMenuItems,
+  color,
+  variant,
+  ariaLabelTriggerButton,
+}: StudioProfileMenuProps): ReactElement => {
+  const [open, setOpen] = useState(false);
+
+  const truncatedText =
```

Comment:
Maybe we should move this out of the component and just pass in the text. It makes the component more flexible 😄